src/components/dashboard/container/DictionariesDisplay.jsx

```diff
@@ -2,28 +2,36 @@ import React, { Component } from 'react';
 import propTypes from 'prop-types';
 import { connect } from 'react-redux';
 import '../styles/index.css';
-import { fetchDictionaries } from '../../../redux/actions/dictionaries/dictionaryActionCreators'
+import { fetchDictionaries, fetchDictionary } from '../../../redux/actions/dictionaries/dictionaryActionCreators'
 import SideBar from '../components/SideNavigation';
 import ListDictionaries from '../components/dictionary/ListDictionaries';
 
 export class DictionaryDisplay extends Component {
+    
     static propTypes = {
         fetchDictionaries: propTypes.func.isRequired,
-        dictionaries: propTypes.arrayOf(propTypes.shape({
+        dictionaries: propTypes.objectOf(propTypes.shape({
             dictionaryName: propTypes.string,
         })).isRequired,
         isFetching: propTypes.bool.isRequired,
     };
     componentDidMount(){
         this.props.fetchDictionaries();
     }
+    handleDictionaryFetch = (data) => {
+        this.props.fetchDictionary(data.url).then(
```

Comment:
You could resolve promises in your action creators, this will improve code maintainability and easier component testing.

Reply:
You could check for the new props in the `cwrp` lifecycle method, then redirect the user after the fetchDictionary method finish executing.

Reply:
would you mind if the fixtures are made in a new Pull-request

Reply:
You could just fix it now